feed/views.py

```diff
@@ -12,4 +14,9 @@ class PostList(generic.ListView):
 
 class PostDetail(generic.DetailView):
     model = BlogPost
-    template_name = 'post_detail.html'
\ No newline at end of file
+    template_name = 'post_detail.html'
+    def get_context_data(self, **kwargs):
+       context = super().get_context_data(**kwargs)
+       context['form'] = CommentForm()
+       context['advert'] = Advert()
```

Comment:
Super close! Try this for getting the first Advert. You can also use `.last()` to get the latest one. Or `.all()` to return a QuerySet (fancy list) of advert objects. 
```suggestion
       context['advert'] = Advert.objects.first()
```